### ISSUE TITLE
[Feature Request] Official Click support (i.e., `@click.command()`) :expressionless: 

### ISSUE THREAD

**Issue by adamtheturtle** [beartype/beartype#442]
Congratulations on the release @leycec . I've found an issue, and a workaround for it.

This works with `beartype==0.18.5` but does not work with `beartype==0.19.0`.

```python
# /// script
# dependencies = ["click==8.1.7", "beartype==0.19.0"]
# ///

import click
from beartype import beartype
from click.testing import CliRunner

@beartype
@click.command()
def main() -> None:
    pass
			
runner = CliRunner()
result = runner.invoke(cli=main)
```

With `beartype==0.19.0` I get:

```
Traceback (most recent call last):
  File ["/Users/adam/Library/Caches/uv/archive-v0/Mj31znXpw5UdBGD5z-rgT/lib/python3.12/site-packages/click/testing.py", line 403](coderunner:///Users/adam/Library/Caches/uv/archive-v0/Mj31znXpw5UdBGD5z-rgT/lib/python3.12/site-packages/click/testing.py#403:0!), in invoke
    prog_name = extra.pop("prog_name")
                ^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'prog_name'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ["/Users/adam/Desktop/bt_example.py", line 17](coderunner:///Users/adam/Desktop/bt_example.py#17:0!), in <module>
    result = runner.invoke(cli=main)
             ^^^^^^^^^^^^^^^^^^^^^^^
  File ["/Users/adam/Library/Caches/uv/archive-v0/Mj31znXpw5UdBGD5z-rgT/lib/python3.12/site-packages/click/testing.py", line 405](coderunner:///Users/adam/Library/Caches/uv/archive-v0/Mj31znXpw5UdBGD5z-rgT/lib/python3.12/site-packages/click/testing.py#405:0!), in invoke
    prog_name = self.get_default_prog_name(cli)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ["/Users/adam/Library/Caches/uv/archive-v0/Mj31znXpw5UdBGD5z-rgT/lib/python3.12/site-packages/click/testing.py", line 195](coderunner:///Users/adam/Library/Caches/uv/archive-v0/Mj31znXpw5UdBGD5z-rgT/lib/python3.12/site-packages/click/testing.py#195:0!), in get_default_prog_name
    return cli.name or "root"
           ^^^^^^^^
AttributeError: 'function' object has no attribute 'name'
```

As well as with click testing, I've seen similar issues with [`sphinx-click`](https://sphinx-click.readthedocs.io/en/latest/).

My workaround is to move from:

```python
@beartype
@click.command()
def main() -> None:
    pass
```

to

```python
@click.command()
@beartype
def main() -> None:
    pass
```

**Comment by RomainBrault** [comment]
I confirm this same error with beartype claw in a click program

**Comment by leycec** [comment]
**Guh!** Technically speaking, Click is ~~almost certainly~~ *probably* to blame here. Click appears to be intolerant of decorator chaining (i.e., of other third-party decorators like `@beartype` preceding the `@click.command()` decorator). But nobody cares about technicalities and neither do I.

Pragmatically speaking, it is genuinely awful that @beartype used to support this and no longer does. I fully agree! Previously, @beartype just kinda ignored Click and hoped for the best. Now, I'll add official support for Click in an upcoming @beartype 0.19.1 patch release in a day or two.

So sorry about this, CLI warriors. @beartype promises to do better for Click shortly. :face_exhaling: 

## Help @Beartype Help Itself

Relatedly, consider installing **@beartype 0.20.0 release candidates** when the time comes in a few months. I tried *really* hard to avoid breaking backward compatibility by [releasing three @beartype 0.19.0 release candidates over the last several months](https://github.com/beartype/beartype/discussions/401). But that only works if everybody installs these release candidates. I guess nobody installed these release candidates, huh? I mean, I *get* it. There's only so much time in the day. But we can avoid pain like this in the future!

Please help @beartype help itself. I beg this of you, fam. :bow:

**Comment by leycec** [comment]
**We'll do this somehow!** This is *probably* the last action item for our upcoming @beartype 0.19.1 patch release. Although praying for a speedy resolution only gets us so far, prior @beartype releases *did* support this. Let us reflect on the past with fondness in our hearts. :face_holding_back_tears:

**Comment by leycec** [comment]
**Resolved by 17e121e33c707dba.** Click and @beartype are now BFFLs once more. Now that @beartype officially supports Click, I'm *fairly* certain that @beartype previously just ignored Click commands; @beartype wasn't actually type-checking Click commands. Now, @beartype type-checks Click commands.

Thanks again for that *super*-awesome minimal working example (MWE), @adamtheturtle. Your clever (and probably intentional) use of the `pytest`-friendly `click.testing.CliRunner` scaffold enabled me to immediately turn this into a working unit test without knowing *anything* about Click. You're awesome! Turtle power forever. :turtle: :fist_raised: :turtle:

**Comment by adamtheturtle** [comment]
Thank you for the great work @leycec ! 🐢 ❤️ 🐻
